### ISSUE TITLE
CS0305 compiler error false trigger?

### ISSUE THREAD

**Issue by CHerSun** [dotnet/roslyn#63053]
**Version Used**: 
Not sure how to check Roslyn version.
DotNet 6.0.302
VS 2022 17.2.6

**Steps to Reproduce**:
Code to check if generic is of supported type, like:

``` C#
type = some_value.GetType();
return type.IsGenericType && (type.GetGenericTypeDefinition() == typeof(List<>));
```
I've checked the code - it does exactly what's expected.

**Expected Behavior**:
My belief is that this check should not trigger CS0305 at all.

**Actual Behavior**:
Reports compiler error CS0305.

**Comment by davidwengier** [comment]
Could you provide more info about where you're seeing the compiler error? That code does not seem to reproduce the issue:

https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLRII5wJZICaoC2EhwECyANHiANQA+AAgEwCMAsAFCMAMABI1YAWANxcujAMwDmfAMJ8A3lz6qB04AHtNAGz4BZABSbgAKwgBjGH2SbiAfQBuUHXAgBKJSrU/nCPjAAngAOEHwAvDZ2EE4ubgB0AOIQMAAqIRCG7mKcPj6MAOwBGfEAksjJAHbk2BbpoXwAZI18hkGhSSlVNXUZACIQYNiV2DDYmpVZEZHtEJpghgAy2MgwADwAfO7Z3qoAvlx7QA=

**Comment by CHerSun** [comment]
@davidwengier I was using a generic with 2 types `Accessor<T1, T2>`. Checking for it I tried to use `typeof(Accessor<>)` - that's where CS0305 was triggered. My bad, that was a correct trigger, sorry that I've wasted your time.

Fixed by using `typeof(Accessor<,>)`.
